components/script/dom/element.rs

```diff
@@ -3773,6 +3773,27 @@ pub fn set_cross_origin_attribute(element: &Element, value: Option<DOMString>) {
     }
 }
 
+pub fn reflect_referrer_policy_attribute(element: &Element) -> DOMString {
+    let attr =
+        element.get_attribute_by_name(DOMString::from_string(String::from("referrerpolicy")));
+
+    if let Some(val) = attr.map(|v| v.Value()) {
+        if val == "" ||
+            val == "no-referrer" ||
+            val == "no-referrer-when-downgrade" ||
+            val == "same-origin" ||
+            val == "origin" ||
+            val == "strict-origin" ||
+            val == "origin-when-cross-origin" ||
+            val == "strict-origin-when-cross-origin" ||
+            val == "unsafe-url"
+        {
+            return val;
+        }
+    }
+    return DOMString::new();
```

Comment:
We need to reflect the attribute case insensitively so the `val` would need to be transformed into lower cases.

I think there are two ways to do this 👀 👇 

1. make the `val` to be mutable and `val.make_ascii_lowercase()` before comparing with those values.

2. use the `match_ignore_ascii_case` macro like `determine_policy_for_token` and then 

```rust
element
  .get_attribute_by_name(DOMString::from_string(String::from("referrerpolicy")))
  .filter(|v| match_ignore_ascii_case! { v.Value(),
    "no-referrer" | "no-referrer-when-downgrade" | "same-origin" |
    "origin" | "strict-origin" | "origin-when-cross-origin" |
    "strict-origin-when-cross-origin" | "unsafe-url" => true,
    _ => false
  })
  .map_or(DOMString::new(), |val| val.to_ascii_lowercase())
```